zerver/views/custom_profile_fields.py

```diff
@@ -57,13 +61,29 @@ def validate_custom_field_data(field_type: int,
     if error:
         raise JsonableError(error)
 
+def is_default_external_field(field_type: int, field_data: ProfileFieldData) -> bool:
+    if field_type != CustomProfileField.EXTERNAL_ACCOUNT:
+        return False
+    if field_data['subtype'] == 'custom':
+        return False
+    return True
+
 def validate_custom_profile_field(name: str, hint: str, field_type: int,
                                   field_data: ProfileFieldData) -> None:
     # Validate field data
     validate_custom_field_data(field_type, field_data)
 
-    # Validate field name, hint and type
-    validate_field_name_and_hint(name, hint)
+    if is_default_external_field(field_type, field_data):
+        # Either name and hint should be empty or equal to default field data
+        default_field = DEFAULT_EXTERNAL_ACCOUNTS[field_data['subtype']]  # type: ignore # key for "Union[Dict[str, str], str]" can be str
+        if name and name != default_field['name']:
+            raise JsonableError(_("Default custom profile field data cannot changed."))
+        if hint and hint != default_field['hint']:
+            raise JsonableError(_("Default custom profile field data cannot changed."))
```

Comment:
This isn't great English, and I'm also not sure how to briefly explain the user's mistake here.  @rishig do you think we need an error for this case, or should we just ignore whatever was set in `name` and use the value from `default_field`?  This isn't a possible situation from the UI, just the API.

Reply:
I don't have a good intuition for the API, but I would say either:
* No error / silently overwrite `name`
* Maybe something like "`name` must be empty for type `external field`"?